From 80e0e9f4e89aca5efde685e6049b88406ca73818 Mon Sep 17 00:00:00 2001
From: yzha108 <yanli.l.zhang@intel.com>
Date: Wed, 12 Sep 2018 01:48:58 +0800
Subject: [PATCH 2267/2302] media: intel-ipu4: ar0231: set exposure's min/max

Message for Open Source:
Correct T1/T2/T3 min/max value in flexible HDR mode

Message for Internal:
[Issue/Feature]
Correct T1/T2/T3 min/max value in flexible HDR mode

[Root Cause/Changes]
Correct T1/T2/T3 min/max value in flexible HDR mode

Change-Id: I679fcd3dac8f47e0a96083b8e11a93b020cdcbef
Signed-off-by: yzha108 <yanli.l.zhang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h b/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
index 3b1fdf6..1905a9e 100644
--- a/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
+++ b/drivers/media/i2c/crlmodule/crl_ar0231at_configuration.h
@@ -2260,8 +2260,8 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = V4L2_CID_EXPOSURE,
 		.name = "T1_COARSE_EXPOSURE_TIME",
 		.type = CRL_V4L2_CTRL_TYPE_INTEGER,
-		.data.std_data.min = 0x0010,
-		.data.std_data.max = 0x0500,
+		.data.std_data.min = 0x0002,
+		.data.std_data.max = 0x04FF,
 		.data.std_data.step = 1,
 		.data.std_data.def = 0x0163,
 		.flags = 0,
@@ -2279,8 +2279,8 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = CRL_CID_EXPOSURE_SHS1,
 		.name = "T2_COARSE_EXPOSURE_TIME",
 		.type = CRL_V4L2_CTRL_TYPE_CUSTOM,
-		.data.std_data.min = 0x0,
-		.data.std_data.max = 0x0500,
+		.data.std_data.min = 0x0002,
+		.data.std_data.max = 0x0300,
 		.data.std_data.step = 1,
 		.data.std_data.def = 0x0002,
 		.flags = V4L2_CTRL_FLAG_UPDATE,
@@ -2299,10 +2299,10 @@ struct crl_v4l2_ctrl ar0231at_v4l2_ctrls[] = {
 		.ctrl_id = CRL_CID_EXPOSURE_SHS2,
 		.name = "T3_COARSE_EXPOSURE_TIME",
 		.type = CRL_V4L2_CTRL_TYPE_CUSTOM,
-		.data.std_data.min = 0x0,
-		.data.std_data.max = 0x0500,
+		.data.std_data.min = 0x0002,
+		.data.std_data.max = 0x0180,
 		.data.std_data.step = 1,
-		.data.std_data.def = 0x0001,
+		.data.std_data.def = 0x0002,
 		.flags = V4L2_CTRL_FLAG_UPDATE,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
-- 
2.7.4

